## Sample #1

I ordered 2 kits from 2 companies (on 3/3/2021):

***Psomagen***: [Gene & GutBiome Combo](https://psomagen.com/gene-gutbiome-combo/) and [GutBiome+](https://psomagen.com/gutbiome/)

***Viome***: [Health Intelligence Test](https://www.viome.com/products/health-intelligence) and [Gut Intelligence Test](https://www.viome.com/products/gut-intelligence)

I very much hope that I can get raw data, but this does provide something that can be compared (even if I can't).

The 2 Psomagen kits arrived on 3/8/2021, and the 2 Viome kits arrived on 3/5/2021.

All 4 samples were collected and mailed on 3/11/2021.  Please note that you may need to use 2 different e-mail addresses in order to compare 2 samples at the same time from Viome.

I recieved my ***Psomagen*** *Gene & GutBiome* report on 3/31/2021.

I recieved my ***Psomagen*** *GutBiome+* report on 4/9/2021.  I think this is a different report than the GutBiome portion of the other report, including diversity scores that are on different scales?  For the same sample, I had a diversity value of 77 for *Gene & GutBiome* and 52 for *GutBiome+*.

The details are also somewhat different:

<table>
  <tbody>
    <tr>
      <th align="center">Genus</th>
      <th align="center">Psomagen<br>Gene & GutBiome</th>
      <th align="center">Psomagen<br>GutBiome+</th>
      <th align="center">Viome<br>Gut Intelligence<br>(downgraded)</th>
      <th align="center">Viome<br>Gut Intelligence<br>(as purchased)</th>
    </tr>
    <tr>
      <td align="center"><i>Agathobacter rectalis</i></td>
      <td align="left">7.6%</td>
      <td align="left">2.3%</td>
      <td align="left">Not Active?</td>
      <td align="left">Not Active?</td>
     </tr>
    <tr>
      <td align="center"><i>Akkermansia muciniphila</i></td>
      <td align="left">3.0%</td>
      <td align="left">1.0%<br>(<i>Akkermansia sp</i>)</td>
      <td align="left">Active</td>
      <td align="left">Active</td>
    </tr>
     <tr>
      <td align="center"><i>Alistipes</i></td>
      <td align="left">4.5%<br>(genus-level)</td>
      <td align="left">10.7%<br>(7 species)</td>
      <td align="left">Active<br>(9 species)</td>
      <td align="left">Active<br>(9 species)</td>
    </tr>
    <tr>
      <td align="center"><i>Bacteroides stercoris</i></td>
      <td align="left">19%</td>
      <td align="left">24.3%</td>
      <td align="left">Active<br>(ATCC 43183)<br>(ATCC CC31F)<br>(ATCC CL09T03C01)<br>(ATCC DSM)</td>
      <td align="left">Active<br>(ATCC 43183)<br>(ATCC CC31F)<br>(ATCC CL09T03C01)<br>(ATCC DSM)</td>
    </tr>
    <tr>
      <td align="center"><i>Bacteroides uniformis</i></td>
      <td align="left">5.5%</td>
      <td align="left">8.2%</td>
      <td align="left">Active</td>
      <td align="left">Active</td>
      </tr>
      <tr>
      <td align="center"><i>Bacteroides vulgatus</i></td>
      <td align="left">17%</td>
      <td align="left">13.8%</td>
      <td align="left">Active<br>(ATCC 8482)</td>
      <td align="left">Active<br>(ATCC 8482)</td>
      </tr>
      <tr>
      <td align="center"><i>Lachnospira eligens</i></td>
      <td align="left">1.5%</td>
      <td align="left">?</td>
      <td align="left">Active?<br>(Lachnospiraceae bacterium?)</td>
      <td align="left">Active?<br>(Lachnospiraceae bacterium?)</td>
    </tr>
    <tr>
      <td align="center"><i>Faecalibacterium prausnitzii</i></td>
      <td align="left">9.2%</td>
      <td align="left">1.9%</td>
      <td align="left">Active</td>
      <td align="left">Active</td>
    </tr>
    <tr>
      <td align="center"><i>Parasutterella excrementihominis</i></td>
      <td align="left">2.0%</td>
      <td align="left">0.6%</td>
      <td align="left">Active<br>(YIT 11859)</td>
      <td align="left">Active<br>(YIT 11859)</td>
      </tr>
      <tr>
      <td align="center"><i>Ruminococcus bicirculans</i></td>
      <td align="left">2.5%</td>
      <td align="left">0.8%</td>
      <td align="left">Active</td>
      <td align="left">Active</td>
      </tr>
      <tr>
      <td align="center"><i>Ruminococcus bromii</i></td>
      <td align="left">2.1%</td>
      <td align="left">?</td>
      <td align="left">Not Active?</td>
      <td align="left">Not Active?</td>
      </tr>
</tbody>
</table>

I think this seems like expected varation (where I would usually want replicates for metagenomics), but I hope that is made clear (and "high" and "low" are different enough to to be less likely for even biological replicates, if these had some from different stool samples - in this case, everything submitted was from ***the same stool sample***).

**This matches what I would expect for 16S biological replicates (even if these are from the same sample, rather than different samples).**  So, this doesn't surprise me, but I hope this is made clear to most customers.  For example, I might check for for **uncommon genera**, and/or changes of **greater than 5%** (or maybe even greater than 10% or 15%) in absolute frequency.  However, Psomagen explained that they are doing ***shotgun metagenomics***, rather than 16S (and the raw data shows noticably more reads for my GutBiome+ sample).  Nevertheless, I think the varialbity might match what I have seen with 16S data.

The Psomagen Gene&GutBiome and Viome (downgraded) Health Intelligence Test come from the same location in the 1 stool.  The Psomagen GutBiome+ and Viome Gut Intelligence Test come from the same location in the stool sample, but it is different than the other 2 samples (even though all 4 came from the same stool sample).

After sending an e-mail request, Psomagen provided me with my raw data on 4/12/2021:

**GutBiome+**:

[http://ngs2.macrogenlab.com/~customer/2104/2103DNHX-0611/C229-J888-E267_R1.fastq.gz](http://ngs2.macrogenlab.com/~customer/2104/2103DNHX-0611/C229-J888-E267_R1.fastq.gz)
[http://ngs2.macrogenlab.com/~customer/2104/2103DNHX-0611/C229-J888-E267_R2.fastq.gz](http://ngs2.macrogenlab.com/~customer/2104/2103DNHX-0611/C229-J888-E267_R2.fastq.gz)

**Gene & GutBiome (*GutBiome*)**:

[http://ngs2.macrogenlab.com/~customer/2104/2103DNMI-0598/B374-R052-T972_R1.fastq.gz](http://ngs2.macrogenlab.com/~customer/2104/2103DNMI-0598/B374-R052-T972_R1.fastq.gz)
[http://ngs2.macrogenlab.com/~customer/2104/2103DNMI-0598/B374-R052-T972_R2.fastq.gz](http://ngs2.macrogenlab.com/~customer/2104/2103DNMI-0598/B374-R052-T972_R2.fastq.gz)

**Gene & GutBiome (*Gene*)**:

[http://ngs2.macrogenlab.com/~customer/2104/2103DNSC-0599/A374-R052-T972.zip](http://ngs2.macrogenlab.com/~customer/2104/2103DNSC-0599/A374-R052-T972.zip)

Viome said that they do not currently provide raw data.

I don't believe that I recieved an e-mail, but I noticed my results were available for my downgraded Viome Health Intelligence Test on 4/15/2021 (and I believe the sample information says that this was available on 4/14/2021).  My regular Gut Intelligence was not ready, as of 4/15/2021.

I can tell that I am not going to avoid a lot of the Viome recommendations to "Avoid" (such as Almonds) or take any of the recommended supplements.

I also added some notes in the table with the Psomagen samples above.

I was able to download my reports for my 2nd Viome sample on 4/26/2021.

I am not currently planning on ordering 2 of each kit for each subsequent sample.

However, for the 1st sample, this also provides the opporunity to compare the scores for the 2 Viome samples **for the same sample**:


<table>
  <tbody>
    <tr>
      <th align="center">Genus</th>
      <th align="center">Viome<br>Gut Intelligence<br>(downgraded)</th>
      <th align="center">Viome<br>Gut Intelligence<br>(as purchased)</th>
    </tr>
    <tr>
      <td align="center">Gut Microbiome Health</td>
      <td align="left">41<br><b>Not Optimal</b></td>
      <td align="left">49<br>Average</td>
     </tr>
    <tr>
      <td align="center">Inflammatory Activity</td>
      <td align="left">41<br>Average</td>
      <td align="left">36<br>Average</td>
     </tr>
    <tr>
      <td align="center">Metabolic Fitness</td>
      <td align="left">28<br>Average</td>
      <td align="left">29<br>Average</td>
     </tr>
    <tr>
      <td align="center">Digestive Efficiency</td>
      <td align="left">56<br>Average</td>
      <td align="left">71<br><i>Good</i></td>
     </tr>
    <tr>
      <td align="center">Gut Lining Health</td>
      <td align="left">67<br>Average</td>
      <td align="left">68<br>Average</td>
     </tr>
    <tr>
      <td align="center">Protein Fermentation</td>
      <td align="left">42<br>Average</td>
      <td align="left">28<br><i>Good</i></td>
     </tr>
    <tr>
      <td align="center">Gas Production</td>
      <td align="left">46<br>Average</td>
      <td align="left">30<br><i>Good</i></td>
     </tr>
    <tr>
      <td align="center">Active Microbial Diversity</td>
      <td align="left">1st Percentile<br><b>Not Optimal</b></td>
      <td align="left">12th Percentile<br>Average</td>
     </tr>
      <tr>
      <td align="center" colspan="3"></td>
     </tr>
    <tr>
      <td align="center">Vegetables to Avoid</td>
      <td align="left">Bell Pepper<br>Broccoli<br>Brussels Sprouts<br>Cabbage<br>Mustard Greens<br>Tomato</td>
      <td align="left">Bell Pepper<br>Tomato</td>
     </tr>
     </tr>
      <td align="center">Proteins and Fats to Avoid</td>
      <td align="left">Almonds<br>Chicken Egg Yolk<br>Pistachios</td>
      <td align="left">Almonds<br>Pistachios</td>
     </tr>
     </tr>
      <td align="center">Fruits and Grains to Avoid</td>
      <td align="left"><i>None</i></td>
      <td align="left"><i>None</i></td>
     </tr>
     </tr>
      <td align="center">Other Food Items to Avoid</td>
      <td align="left"><i>None</i></td>
      <td align="left"><i>None</i></td>
     </tr>
</tbody>
</table>

So, I think it is fair to say that you should probably **hesitate to take any action based upon these results**.  I expected this for most values with thresholds that I would guess were within a normal distribution.  However, I was a bit surprised by the 1st to 12th percentile difference, even though the absolute difference is perhaps comparible to other categories (where I would guess I should probably be "average" for most or all of them).

## Sample #2

I ordered 3 kits from 3 companies on 4/26/2021 (1 for each program):

***Psomagen***: [GutBiome+](https://psomagen.com/gutbiome/)

***Viome***: [Gut Intelligence Test](https://www.viome.com/products/gut-intelligence)

***thryve***: [Gut Health Test](https://www.thryveinside.com/products/thryves-gut-health-test)

The Viome and thryve kits arrived on 4/29/2021.

My Psomagen kit arrived on 4/30/2021.

All 3 samples collected on 5/3/2021.

**I did not make any changes to my diet, based upon the previous set of results.**

I recieved my **thryve** results on 5/19/2021.  This includes CSV assignments for classifications, as well as FASTQ files.

I received a notification that my *Viome* sample would need to be reprocessed (same sample, new sequencing run) on 5/27/2021.

I recieved by **Psomagen** report on 5/29/2021.

Similar to before, here is a comparison of classificaiton for this same sample:

<table>
  <tbody>
    <tr>
      <th align="center">Genus</th>
      <th align="center">Psomagen<br>GutBiome+</th>
      <th align="center">Viome<br>Gut Intelligence<br>(as purchased)</th>
      <th align="center">thryve</th>
    </tr>
    <tr>
      <td align="center"><i>Agathobacter rectalis</i></td>
      <td align="left">2.2%</td>
      <td align="left">Not Active?</td>
      <td align="left">Not Classified?</td>
     </tr>
    <tr>
      <td align="center"><i>Akkermansia sp</i></td>
      <td align="left">1.4%</td>
      <td align="left">Active</td>
      <td align="left">0.916%<br>(genus-level)</td>
    </tr>
     <tr>
      <td align="center"><i>Alistipes</i></td>
      <td align="left">13.2%<br>7 species</td>
      <td align="left">Active<br>9 species</td>
      <td align="left">3.223%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Bacteroides stercoris</i></td>
      <td align="left">26.5%</td>
      <td align="left">Active<br>4 strains</td>
      <td align="left">54.456%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Bacteroides uniformis</i></td>
      <td align="left">5.1%</td>
      <td align="left">Active</td>
      <td align="left">see above</td>
      </tr>
      <tr>
      <td align="center"><i>Bacteroides vulgatus</i></td>
      <td align="left">9.1%</td>
      <td align="left">Active</td>
      <td align="left">see above</td>
      </tr>
      <tr>
      <td align="center"><i>Bifidobacterium</i></td>
      <td align="left">Not Detected</td>
      <td align="left">Active<br>5 subsp / strains</td>
      <td align="left">0.026%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Blautia</i></td>
      <td align="left">Not Reported?</td>
      <td align="left">Active<br>14 species / strains</td>
      <td align="left">3.949%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Eubacterium</i></td>
      <td align="left">Not Reported?</td>
      <td align="left">Active<br>7 species / strains</td>
      <td align="left">6.802%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Faecalibacterium prausnitzii</i></td>
      <td align="left">2.0%</td>
      <td align="left">Active</td>
      <td align="left">1.687%<br>(genus-level)</td>
    </tr>
    <tr>
      <td align="center"><i>Lactobacillus</i></td>
      <td align="left">Not Detected</td>
      <td align="left">Active<br>8 species / strains</td>
      <td align="left">0.006%<br>(genus-level)</td>
      </tr>
    <tr>
      <td align="center"><i>Parasutterella excrementihominis</i></td>
      <td align="left">0.9%</td>
      <td align="left">Active</td>
      <td align="left">0.011%</td>
      </tr>
    <tr>
      <td align="center"><i>Pseudomonas</i></td>
      <td align="left">Not Reported?</td>
      <td align="left">Active<br>2 species</td>
      <td align="left">0.002%<br>(genus-level)<br>(bad bacteria,<br> healthy level)</td>
      </tr>
    <tr>
      <td align="center"><i>Roseburia</i></td>
      <td align="left">Not Reported?</td>
      <td align="left">Active<br>6 species / strains</td>
      <td align="left">0.717%<br>(genus-level)</td>
      </tr>
    <tr>
      <td align="center"><i>Ruminococcus bicirculans</i></td>
      <td align="left">0.3%</td>
      <td align="left">Active<br>6 species</td>
      <td align="left">6.734%<br>(genus-level)</td>
      </tr>
</tbody>
</table>

I sent an e-mail to **Psomagen** to obtain my FASTQ files on 5/29/2021, and I recieved links to download the FASTQ files on 6/8/2021:

[http://ngs2.macrogenlab.com/~customer/2106/2105DNHX-0230/C058-S779-Y072_R1.fastq.gz](http://ngs2.macrogenlab.com/~customer/2106/2105DNHX-0230/C058-S779-Y072_R1.fastq.gz)

[http://ngs2.macrogenlab.com/~customer/2106/2105DNHX-0230/C058-S779-Y072_R2.fastq.gz](http://ngs2.macrogenlab.com/~customer/2106/2105DNHX-0230/C058-S779-Y072_R2.fastq.gz)

I recieved my **Viome** results on 6/16/2021.

## Other Notes

**1)** There is no liquid in the Psomagen stool collection tubes, so I don't believe there is anything to stop or reduce bacterial growth post-collection.

Post-collection bacterial growth includes things like the "[blooms](https://msystems.asm.org/content/3/3/e00031-18)" discussed in the American Gut project paper.

It looks like the blog post that caused me to design this experiement with newer metagenomics samples is [now private](https://twitter.com/UseqMiseq/status/466559572069453825).  However, I believe that also tested technical replicates submitted to multiple companies.

**2)** I did not realize it at the time of ordering, but one of the Viome kits (including Viome Health Intelligence) requires a **blood** sample.  To be fair, this is listed on the [description](https://www.viome.com/products/health-intelligence) for the product, if you scroll down or find the word "Collection Method"

To help me remember (and  perhaps make more clear to others) here is the content related to the blood collection:

"***Fasting: Viome requires fasting before collecting your Human Gene Expression™ (blood) sample.** The simplest way to do this is to collect your sample right after you wake, before eating any food or drinking anything other than water (Note—your kit instructions may not contain this step, please disregard and follow the directions here).*

*Watch our [step-by-step instruction video here](https://www.viome.com/stepstotest/health-intelligence/collect-hi) for the best techniques for easy collection.*

*We recommend **collecting both your blood and stool samples within the same day** (if this is too difficult for you, a few days in between samples will not impact your results).*"

My experiences with my [nutrigenomics](https://github.com/cwarden45/DTC_Scripts/tree/master/Nutrigenomics) testing (which included some metabolite measurements) was that I don't think I prefer the at-home collection (if it can be avoided).  In fact, if you watch that video, you will see that you need ***4 Minivettes*** of blood (each of which appears to require multiple drops of blood, based upon 2:30 to 2:55 in the video).  For this reason, I only submitted the stool sample for my Health Intelligence Test (converting my kit type, without requiring blood collection).

I am not concerned about the extra costs for the part that I do not want to collect (the blood samples).  However, others might want to know that a partial refund is not possible (if they find themselves in a similar situation).  For me, this was only an extra $100 with the sale price.  However, I think the regular price difference is $300 (as of 3/10/2021).

**3)** It might also be worth noting that [Psomagen](https://psomagen.com/) provides other genomics services, but I believe the others are not Direct-To-Consumer.

**4)** I noticed that Nebula provides reviews for [Psomagen](https://nebula.org/blog/psomagen-review/), [Viome](https://nebula.org/blog/viome-review/), and [Thryve](https://nebula.org/blog/thryve-review/).  While I don't agree with everything, **the Viome review includes other related companies at the end**.  Also, to be fair, the Viome ratings seem high, but the review also says "*The validity of the Viome results provided are questionable*" and repeats that the results should not be used for medical decisions (even though there are explicit supplement recommendations from the company).  Likewise, the ratings for Thryve seem high to me, given the review says things like "*Much of the microbiome analysis seems tailored to selling their own supplements*" and "*Some customers find supplements cause stomach discomfort*" (where I think the later is arguably literally a description of some amount of harm being caused by the product's recommendation - or, if the individual was not already healthy, then they may benefit from seeing a health care professional).

I double-checked with the vendor and the [Floré Gut Health Test Kit](https://sungenomics.com/products/microbiome-collection-kit) from Sun Genomics does **not** offer raw FASTQ files.

My understanding is that the [BIOHM Gut Test](https://www.biohmhealth.com/products/gut-report-kit-new) does not currently provide FASTQ files, but they would be willing to work with me to provide "raw" data individually (at least for Excel files with read classification counts).  So, I think I will probably check that out for my 3rd sample.
